pkg/api/networkservice/mechanisms/kernel/helpers.go

```diff
@@ -87,21 +85,7 @@ func (m *Mechanism) IsPCIDevice() bool {
 }
 
 // GetInterfaceName returns the Kernel Interface Name
-//                  this is Mechanism.Parameters[InterfaceNameKey] if set
-//                  otherwise returns a name computed from networkservice.Connection 'conn'
-func (m *Mechanism) GetInterfaceName(conn *networkservice.Connection) string {
-	if m.GetParameters()[InterfaceNameKey] == "" {
-		ns := conn.GetNetworkService()
-		nsMaxLength := LinuxIfMaxLength - 5
-		if len(ns) > nsMaxLength {
-			ns = ns[:nsMaxLength]
-		}
-		name := fmt.Sprintf("%s-%s", ns, conn.GetId())
-		if len(name) > LinuxIfMaxLength {
-			name = name[:LinuxIfMaxLength]
-		}
-		m.GetParameters()[InterfaceNameKey] = name
-	}
+func (m *Mechanism) GetInterfaceName() string {
 	name := m.GetParameters()[InterfaceNameKey]
```

Comment:
Nit: should we throw error if `name` is empty ? but it might break backward compatibility. but anyway name won't be empty if this method is used appropriately.